src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,51 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int amountOfAddedObjects;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[STORAGE_SIZE];
+        this.values = (V[]) new Object[STORAGE_SIZE];
+        this.amountOfAddedObjects = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (keyIndex(key) > -1 && keyCompare(key, keys[keyIndex(key)])) {
```

Comment:
```suggestion
        if (keyIndex(key) > -1) {
```
I guess this conditional is redundant